recipes/premake/5.x/conanfile.py

```diff
@@ -1,55 +1,86 @@
-from conans.errors import ConanInvalidConfiguration
 import glob
 import os
 import re
+import shutil
+
+from conan import ConanFile, conan_version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.files import apply_conandata_patches, chdir, copy, export_conandata_patches, get, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import MSBuild, MSBuildToolchain, is_msvc, check_min_vs
+
+required_conan_version = ">=1.53.0"
 
 
 class PremakeConan(ConanFile):
     name = "premake"
-    topics = ("conan", "premake", "build", "build-systems")
-    description = "Describe your software project just once, using Premake's simple and easy to read syntax, and build it everywhere"
+    description = (
+        "Describe your software project just once, "
+        "using Premake's simple and easy to read syntax, "
+        "and build it everywhere"
+    )
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://premake.github.io"
-    license = "BSD-3-Clause"
+    topics = ("build", "build-systems")
+
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
-    exports_sources = "patches/**"
     options = {
         "lto": [True, False],
     }
     default_options = {
         "lto": False,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
-        if self.settings.os != "Windows" or self.settings.compiler == "Visual Studio":
-            del self.options.lto
+        if self.settings.os != "Windows" or is_msvc(self):
+            self.options.rm_safe("lto")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def requirements(self):
+        if self.settings.os != "Windows":
+            self.requires("util-linux-libuuid/2.39")
 
     def validate(self):
-        if hasattr(self, 'settings_build') and tools.cross_building(self, skip_x64_x86=True):
+        if hasattr(self, "settings_build") and cross_building(self, skip_x64_x86=True):
             raise ConanInvalidConfiguration("Cross-building not implemented")
+        if conan_version.major == 1 and self.settings.build_type == "Debug":
+            # https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/18844/15-linux-clang/premake/5.0.0-alpha15/summary.json
+            raise ConanInvalidConfiguration("Debug build not supported with Conan 1.x")
```

Comment:
It's not clear to me why this fails. Also, why does it fail in conan 1 and not in conan 2? Also, please don't use that link for the logs, it could disappear at any time

Reply:
I cannot reproduce this locally and I suppose it might be possible to reproduce it in a container, but the Debug + Conan v1 combination is too niche to be worth the effort, IMO.